### PR TITLE
Added kms CMK support for ecs

### PR DESCRIPTION
**Pull request by mayank0202**
## Description

Added kms CMK support for ecs

## Related Issue

- [Issue Number](link-to-issue): Brief description

## Checklist

Please ensure that the following steps are completed before submitting the pull request:

- [x] Code follows the Terraform best practices and style guidelines.
- [ ] Changes are appropriately documented, including any necessary updates to README or other documentation files.
- [ ] Unit tests have been added or updated to cover the changes introduced by this pull request.
- [x] Changes have been tested locally and verified to work as expected.
- [x] The code has been reviewed to ensure it aligns with the project's goals and standards.
- [x] Dependencies and backward compatibility have been considered and addressed if applicable.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Testing Instructions

Ran example folder and it deployed all resources perfectly , nothing is a breaking change

### CODE REVIEW

**Comment by mayank0202** [comment]
@tsaucier-sf  as per discussion we don't have functionality for  ECS Fargate ephemeral storage

[issues-aws-provider](https://github.com/hashicorp/terraform-provider-aws/issues/37922)

Currently only `execute_command_configuration` is allowed in an ECS cluster configuration in Terraform which i have added  in this PR

File: variables.tf [hunk 1]
```diff
@@ -217,3 +217,36 @@ variable "additional_ssm_params" {
   EOF
   default     = []
 }
+
+variable "enabled" {
```
Comment (vijay-stephen) [inline variables.tf:221]: This variable is not required. instead use `kms_key_arn` defaulted to null

File: variables.tf [hunk 2]
```diff
@@ -217,3 +217,36 @@ variable "additional_ssm_params" {
...
+  description = "Flag to determine whether to create a KMS key for the ECS cluster."
+  type        = bool
+  default     = false
+}
+
+variable "ecs_kms_key_arn_override" {
+  description = "Override ARN for the ECS KMS key."
+  type        = string
+  default     = ""
+}
```
Comment (vijay-stephen) [inline variables.tf:231]: Use `kms_key_arn` defaulted to null,

null / non-null to enable disable

File: main.tf [hunk 3]
```diff
@@ -33,6 +34,16 @@ module "ecs" {
     Name = local.cluster_name
   }))
 }
+module "kms" {
+  source                  = "sourcefuse/arc-kms/aws"
+  version                 = "1.0.0"
+  enabled                 = var.enabled
```
Comment (vijay-stephen) [inline main.tf:40]: please don't use this variable.

**Review by vijay-stephen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**